tests/invalid/Earthfile

```diff
@@ -20,18 +20,24 @@ WORKDIR /test
 test-trailing-backslash:
     DO +RUN_EARTHLY_ARGS --earthfile=trailing-backslash.earth --should_fail=true --target=+base
 
+test-leading-whitespace:
+    DO +RUN_EARTHLY_ARGS --earthfile=leading-whitespace.earth --should_fail=true --target=+base --output_contains="syntax error"
```

Comment:
this feels incorrect,

```
VERSION 0.6
FROM alpine
ARG foo
RUN echo foo is $foo
```

should be equivalent to 

```
VERSION 0.6
                            FROM alpine
                            ARG foo
                            RUN echo foo is $foo
```

which are both valid Earthfiles, which should output:

```
$ earthly --no-cache  +base --foo=bar
 1. Init 🚀
————————————————————————————————————————————————————————————————————————————————

           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)

 2. Build 🔧
————————————————————————————————————————————————————————————————————————————————

              alpine | --> Load metadata alpine linux/amd64
               +base | --> FROM alpine
               +base | [----------] 100% resolve docker.io/library/alpine@sha256:f271e74b17ced29b915d351685fd4644785c6d1559dd1f2d4189a5e851ef753a
               +base | foo=bar
               +base | --> RUN echo foo is $foo
               +base | foo is bar
              output | --> exporting outputs

 3. Push ⏫ (disabled)
————————————————————————————————————————————————————————————————————————————————

To enable pushing use earthly --push

 4. Local Output 🎁
————————————————————————————————————————————————————————————————————————————————



========================== 🌍 Earthly Build  ✅ SUCCESS ==========================
```

Reply:
I did originally write the test to just `grep -v panic earthly.output`, but that was passing even when earthly was panicking. It felt like I was going to have to make a lot more changes to get that to work.  I'll revisit that though. There are probably other tests we'll want to use that for, so it's probably worth doing.

FWIW, since the `Earthfile` language relies on significant whitespace at the top level, I honestly think that erroneous indentation _should_ be a syntax error (again, just at the top level).

We only have one level of significant whitespace - i.e. our parser grammar only ever cares whether or not there is indentation, never how much indentation there is. Indenting more or less doesn't change anything once you've started an `INDENT`. But it makes sense that leading whitespace at the top level would trigger an `INDENT` token, and that an unexpected `INDENT` token would cause parser errors.

Reply:
> I honestly think that erroneous indentation should be a syntax error (again, just at the top level).

My mistake! for some reason I thought this was valid,

```
foo:
FROM alpine
RUN echo "this will cause a syntax error since there's no leading whitespace"
```